v3/lints/cabf_br/lint_underscore_present_with_too_long_validity.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package cabf_br
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+func init() {
+	lint.RegisterLint(&lint.Lint{
+		Name:            "e_underscore_present_with_too_long_validity",
+		Description:     "From December 10th 2018 to April 1st 2019 DNSNames may contain underscores if-and-only-if the certificate is valid for less than thirty days.",
+		Citation:        "BR 7.1.4.2.1",
+		Source:          lint.CABFBaselineRequirements,
+		EffectiveDate:   util.CABFBRs_1_6_2_Date,
+		IneffectiveDate: util.CABFBRs_1_6_2_UnderscorePermissibilitySunsetDat,
+		Lint:            func() lint.LintInterface { return &UnderscorePresentWithTooLongValidity{} },
+	})
+}
+
+type UnderscorePresentWithTooLongValidity struct{}
+
+func (l *UnderscorePresentWithTooLongValidity) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && util.DNSNamesExist(c)
```

Comment:
Can/should the check from 47 - 49 be moved here?

```suggestion
	return util.IsSubscriberCert(c) &&
	       util.DNSNamesExist(c) &&
	       c.NotBefore.AddDate(0, 0, 30).Before(c.NotAfter)
```

Might be a way to simplify further. But is a bit of a suggestion

Reply:
I don't think so? If we do this check here then what we would be saying is that this lint does not apply to this certificate at all (which is usually reserved for scenarios such as being an irrelevant cert type or not having a particular field).

I do prefer that method of adding 30 days, though. Far more correct than the naive `time.Hour*24*30`.